aiohue/v2/controllers/events.py

```diff
@@ -0,0 +1,191 @@
+            resource_filter = (resource_filter,)
+        subscription = (callback, event_filter, resource_filter)
+
+        def unsubscribe():
+            self._subscribers.remove(subscription)
+
+        self._subscribers.append(subscription)
+        return unsubscribe
+
+    def emit(self, type: EventType, data: CLIPResource) -> None:
+        """Emit event to all listeners."""
+        for (callback, event_filter, resource_filter) in self._subscribers:
+            if event_filter is not None and type not in event_filter:
+                continue
+            if resource_filter is not None and data.type not in resource_filter:
+                continue
+            if iscoroutinefunction(callback):
+                asyncio.create_task(callback(type, data))
+            else:
+                callback(type, data)
+
+    async def __event_reader(self) -> NoReturn:
+        """
+
+        Read incoming SSE messages and put them in a Queue to be processed.
+
+        Background task that keeps (re)connecting untill stopped.
+
+        https://html.spec.whatwg.org/multipage/server-sent-events.html#event-stream-interpretation
+        """
+        self._status = EventStreamStatus.CONNECTING
+        headers = {"Accept": "text/event-stream", "Cache-Control": "no-cache"}
+        retries = 0
+        while True:
+            if self._last_event_id:
+                headers["Last-Event-ID"] = self._last_event_id
+            try:
+                async with self._bridge.create_request(
+                    "get", "eventstream/clip/v2", timeout=0, headers=headers
+                ) as resp:
+                    # update status to connected once we reach this point
+                    self._status = EventStreamStatus.CONNECTED
+                    retries = 0  # reset on succesfull connect
+                    self._logger.debug("Connected to EventStream")
+                    # messages come in one by line, according to EventStream/SSE specs
+                    # we iterate over the incoming lines in the streamreader.
+                    # To prevent a deadlock waiting for a message while the connection is dead
+                    # we have a simple timeout guard in place.
+                    # If no message is received in 30 minutes, a Timeout error will be raised
+                    # and thus the connection re-established.
+                    iterator = resp.content.__aiter__()
+                    while True:
+                        line = await asyncio.wait_for(
+                            iterator.__anext__(), timeout=30 * 60
+                        )
+                        self.__parse_message(line)
+            except (ClientConnectionError, asyncio.TimeoutError) as err:
+                status = getattr(err, "status", None)
+                if status is not None and status == 404:
+                    raise InvalidAPIVersion from err
+                if status is not None and status == 403:
```

Comment:
No need to check if not equal to X and equal to Y. Equal to Y implies not equal to X.
```suggestion
                if status == 404:
                    raise InvalidAPIVersion from err
                if status == 403:
```